### PR TITLE
Dockerized Version

### PR DESCRIPTION
**Pull request by Ans-Saeed**
Dear Chintan Golakiya,

I would like to commend you on the excellent job you did creating your nodejs app. Recently, I took the liberty of forking your repository and adding a Dockerized version to it. This new version makes it very easy for someone who is new to the app to get started locally. All they need to do is navigate to the main project directory and run two simple commands:

1. docker-compose build
2. docker-compose up

After executing these commands, the app will be up and running in no time, and it can be accessed locally using the following ports:

- Server: localhost/5000
- Frontend: localhost/3100
- User-portal: localhost/3200
- Mongo-express: localhost/8080

This improvement will make it much easier for new users to get your app up and running quickly after a git pull/clone.

Thank you for taking the time to read my message, and I look forward to hearing from you.

Best regards,
Ans Saeed

### CODE REVIEW

**Comment by chintan-golakiya** [comment]
Hi Ans-saeed,

Thank you for creating a dockerized version and sharing with pull request.
I would like to add following points

- can you please keep "backend" folder name as it is? I do not see any reason to rename it to "server"
- I tried to run "docker-compose build" and "docker-compose up" with your code changes, but it shows error for database connection from server app to mongodb
- please use password  for admin user with hash function because without hash function, authentication will fail.

Thank you,
Chintan Golakiya

**Comment by Ans-Saeed** [comment]
Dear Chintan,

Thank you I see you have added Docker files and Docker Compose files to your project. Although I am not a developer, I have recently acquired knowledge in Docker and have attempted to convert your project to a Dockerized version. Unfortunately, I was unable to connect using admin as I am not familiar with how JWT or hashcode functions in your code. Nonetheless, I am pleased to see that you were able to update your code to a Dockerized version yourself.

I would like to request that you add me as a contributor to your project. This will greatly benefit my GitHub profile and increase my chances of securing a job in the DevOps field.

Please let me know further how can i contribute to your project.

Thank you.

Best regards,
Ans Saeed

File: user-portal-frontend/Dockerfile
Comment (chintan-golakiya) [inline user-portal-frontend/Dockerfile:1]: WORKDIR should be /usr/user-portal-frontend in user-portal-frontend/Dockerfile.

**Review by chintan-golakiya — COMMENTED** [review]
for user-portal-frontend/Dockerfile, change is not needed on line 3. 
please check.

File: user-portal-frontend/Dockerfile
Comment (Ans-Saeed) [reply user-portal-frontend/Dockerfile:1]: yes thanks for noticing I corrected it now.

**Comment by Ans-Saeed** [comment]
Thankyou very much this means alot this is my first contribution. :)
May you be blessed with abundant success in all your future endeavors.

**Pull request merged**